test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -5158,36 +5239,121 @@ def save_configuration(self, outfile, mode='a'):
         """
         conf = {}
         sconf = {MGR_OBJ_SERVER: conf}
-
-        rd = os.path.join(self.pbs_conf['PBS_HOME'], 'server_priv',
-                          'resourcedef')
-        self._save_config_file(conf, rd)
-
+        # save pbs.conf file
+        cfg_path = self.du.get_pbs_conf_file()
+        with open(cfg_path, 'r') as p:
+            pbs_cfg = p.readlines()
+            config = self.utils.convert_to_dictlist(pbs_cfg)
+            pbs_cfg_b64 = base64.encodestring(str(config[0]))
+        conf['pbs_conf'] = pbs_cfg_b64
+        # save hook files
+        hooks_str = self._save_hook_files()
+        conf.update(hooks_str)
+        conf['hooks'] = hooks_str
         qmgr = os.path.join(self.client_conf['PBS_EXEC'], 'bin', 'qmgr')
-
-        ret = self.du.run_cmd(self.client, [qmgr, '-c', 'p s'], sudo=True)
+        pbsnodes = os.path.join(
+            self.client_conf['PBS_EXEC'], 'bin', 'pbsnodes')
+        ret = self.du.run_cmd(
+            self.hostname, [
+                qmgr, '-c', 'print server'], sudo=True,
+            logerr=False, level=logging.DEBUG)
         if ret['rc'] != 0:
             return False
         else:
             conf['qmgr_print_server'] = ret['out']
-
-        ret = self.du.run_cmd(self.hostname, [qmgr, '-c', 'p sched'],
-                              sudo=True)
+        ret = self.du.run_cmd(self.hostname, [qmgr, '-c', 'print sched'],
+                              logerr=False, level=logging.DEBUG, sudo=True)
         if ret['rc'] != 0:
             return False
         else:
             conf['qmgr_print_sched'] = ret['out']
+        ret = self.du.run_cmd(self.hostname, [pbsnodes, '-av'],
+                              logerr=False, level=logging.DEBUG)
+        nodes_val = self.utils.convert_to_dictlist(ret['out'])
+        conf['pbsnodes'] = nodes_val
+        try:
+            with open(outfile, mode) as f:
+                json.dump(conf, f)
+        except BaseException:
+            self.logger.error('Error processing file ' + outfile)
+            return False
 
-        ret = self.du.run_cmd(self.hostname, [qmgr, '-c', 'p h'], sudo=True)
+        return True
+
+    def _save_hook_files(self):
+        """
+        save all the hooks .CF, .PY, .HK files
+        """
+        qmgr = os.path.join(self.client_conf['PBS_EXEC'], 'bin', 'qmgr')
+        cfg = {"hooks": ""}
+        cmd = [qmgr, '-c', 'print hook @default']
```

Comment:
Just have a doubt here. @default is for specifying server right? If we do `print hook` then that should also return the same output. Why are we using @default explicitly?

Reply:
As per admin guide: @default refers to default server.